lmnet/lmnet/networks/classification/vgg7.py

```diff
@@ -0,0 +1,268 @@
+            num_outputs=filters,
+            weights_initializer=kernel_initializer,
+            biases_initializer=biases_initializer,
+            activation_fn=activation,
+        )
+
+        return output
+
+    def convert_rbg_to_bgr(self, rgb_images):
+        red, green, blue = tf.split(axis=3, num_or_size_splits=3, value=rgb_images)
+
+        bgr_images = tf.concat(axis=3, values=[
+            blue - VGG_MEAN[0],
+            green - VGG_MEAN[1],
+            red - VGG_MEAN[2],
+        ])
+
+        return bgr_images
+
+    def max_pool(
+            self,
+            name,
+            inputs,
+            kernel_size=2,
+            strides=2
+    ):
+        if self.data_format == 'NHWC':
+            data_format = 'channels_last'
+        else:
+            data_format = 'channels_first'
+
+        output = tf.layers.max_pooling2d(
+            name=name,
+            inputs=inputs,
+            pool_size=kernel_size,
+            strides=strides,
+            data_format=data_format,
+        )
+
+        return output
+
+
+class Vgg7Quantize(Vgg7Network, BaseQuantize):
+    version = 1.0
+
+    def __init__(
+            self,
+            quantize_first_convolution=True,
+            activation_quantizer=None,
+            activation_quantizer_kwargs=None,
+            weight_quantizer=None,
+            weight_quantizer_kwargs=None,
+            *args,
+            **kwargs
+    ):
+        Vgg7Network.__init__(
+            self,
+            *args,
+            **kwargs
+        )
+
+        BaseQuantize.__init__(
+            self,
+            activation_quantizer,
+            activation_quantizer_kwargs,
+            weight_quantizer,
+            weight_quantizer_kwargs,
+            quantize_first_convolution,
+        )
+#        self.custom_getter = functools.partial(self._quantized_variable_getter,
+#                                               quantize_first_convolution=self.quantize_first_convolution,
+#                                               weight_quantization=self.weight_quantization)
+
+    @staticmethod
+    def _quantized_variable_getter(getter, name, quantize_first_convolution, weight_quantization=None, *args, **kwargs):
+        """Get the quantized variables.
+
+        Use if to choose or skip the target should be quantized.
+
+        Args:
+            getter: Default from tensorflow.
+            name: Default from tensorflow.
+            weight_quantization: Callable object which quantize variable.
+            args: Args.
+            kwargs: Kwargs.
+        """
+
+        assert callable(weight_quantization)
+        var = getter(name, *args, **kwargs)
+        with tf.variable_scope(name):
+            # Apply weight quantize to variable whose last word of name is "kernel".
+            if not quantize_first_convolution:
+                if var.op.name.startswith("conv1/"):
+                    return var
+
+            if "kernel" == var.op.name.split("/")[-1]:
```

Comment:
I'm not sure, fully connect weight are quantized.

Reply:
I look at .pb file and found that the fully connected layer is not quantized.
<img width="877" alt="screen shot 2018-12-06 at 18 07 03" src="https://user-images.githubusercontent.com/43122237/49573592-be15d780-f981-11e8-8ddc-0fdba6443438.png">